roles/matrix-synapse/tasks/ext/appservice-irc/setup.yml

```diff
@@ -0,0 +1,73 @@
+---
+- name: Ensure Appservice IRC image is pulled
+  docker_image:
+    name: "{{ matrix_appservice_irc_docker_image }}"
+  when: "matrix_appservice_irc_enabled"
+
+- name: Ensure Appservice IRC configuration path exists
+  file:
+    path: "{{ matrix_appservice_irc_base_path }}"
+    state: directory
+    mode: 0750
+    owner: "{{ matrix_user_username }}"
+    group: "{{ matrix_user_username }}"
+  when: "matrix_appservice_irc_enabled"
+
+- name: Ensure Matrix Appservice IRC config installed
+  copy:
+    content: "{{ matrix_appservice_irc_configuration|to_nice_yaml }}"
+    dest: "{{ matrix_appservice_irc_base_path }}/config.yaml"
+    mode: 0644
+    owner: "{{ matrix_user_username }}"
+    group: "{{ matrix_user_username }}"
+  when: "matrix_appservice_irc_enabled"
+
+- stat:
+    path: "{{ matrix_appservice_irc_base_path }}/passkey.pem"
+  register: irc_passkey_file
+
+- name: Generate matrix-appservice-irc passkey if it doesn't exist
+  shell: /usr/bin/openssl genpkey -out {{ matrix_appservice_irc_base_path }}/passkey.pem -outform PEM -algorithm RSA -pkeyopt rsa_keygen_bits:2048
```

Comment:
We can probably do this by not shelling out to the `openssl` binary, but rather by using Ansible's [openssl_privatekey](https://docs.ansible.com/ansible/latest/modules/openssl_privatekey_module.html) module.

I haven't tested it, but it should be something like this, I imagine:

```
- name: Generate matrix-appservice-irc passkey if it doesn't exist
  openssl_privatekey:
    path: "{{ matrix_appservice_irc_base_path }}/passkey.pem"
    size: 2048
    force: false
  when: matrix_appservice_irc_enabled
```
Note: it doesn't even need a `irc_passkey_file.stat` stuff, because the module only regenerates the file if it doesn't exist. So the above `stat` call for `irc_passkey_file` can probably be removed.

However, I'm not sure if using the `openssl_privatekey` is an improvement.
It requires the python-pyOpenSSL library, which I'm not sure if it's available by default.

I guess it may be safer to go with the `openssl` binary and keep the existing code. If you'd like to experiment and find out if some major distros can execute `openssl_privatekey` without explicitly installing additional packages, you could.. If it's too much work, it's good to leave it as it is.

Reply:
I'd say that it's a better idea to use the built-in module instead of calling out to binaries. I tried it on my machine (Arch) and it worked just fine (I already had the python-pyopenssl package install). I'll add a note to the doc page that says it's required to use the IRC bridge.